### PR TITLE
feat: v2 LTS Release

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
Moderate update of react-first-router.

This is an early version of `rudy` which will be superseded by `rudy-respond` which is a complete rewrite.

This release is designed to being a handful of fixes to redux-first-router, while attempting to keep the api the same as well as suppress breaking changes. 

Too much has changed in here, and before my time on this project. The docs will be your friend on this one! 

Fixes: #276 #285

### CODE REVIEW

**Comment by ScriptedAlchemy** [comment]
@hedgepigdaniel @faceyspacey 

Any additional action items? Besides on last read over the codebase - i think this release is ready

**Comment by hedgepigdaniel** [comment]
Looks good to me - its essentially `@rudy` from NPM (which I've been using in production for a long time with no issues). There's a few known bugs but I don't think this introduces any new ones, at least not in the existing feature set.

Only thing I'd suggest is to change the version number

https://github.com/faceyspacey/redux-first-router/blob/509d7bc5906e61d8a6aba61263bbb41725dcfb4c/package.json#L3-L3

If we're leaving `master` and `next` behind, how about `2.0.0`? That way it will be the highest version number on NPM (so it will be obvious that its the latest version), and it will be a major version bump (which it should be, since there are breaking changes compared to previous releases other than `@rudy`).

**Comment by hedgepigdaniel** [comment]
Fixing the CI failures is probably a good idea for future maintenance, but I think we can do that later.

**Comment by ScriptedAlchemy** [comment]
For sure, bumping the version - still waiting on npm access. 


Known bugs, is it anything we can fix easily? Or Rudy Respond kinda stuff?

**Comment by hedgepigdaniel** [comment]
https://github.com/faceyspacey/redux-first-router/issues/285, https://github.com/faceyspacey/redux-first-router/issues/283, https://github.com/faceyspacey/redux-first-router/issues/276

They all sound easy to fix. I don't think any of them are regressions between `next` and `rudy`, but potentially some are regressions from `master` to `rudy`

**Comment by ScriptedAlchemy** [comment]
Okay I’ll see if I can start fixing some of those. 

Also, I got a reply from Dan regarding Suspense whatcha think about us trying out react 17 and it’s async magic on Rudy Respond. It could save us much time and it might be a good move to gear Rudy Respond towards suspense. Either on another branch or the current. 

I like the idea of handing it over to react. Plus we would be very much ahead of the curve if we started assessing the compatibility of suspense and Rudy Respond. 

I’m planning to tinker around as soon as I’ve got a moment.

**Comment by hedgepigdaniel** [comment]
Cool, me too. Snyk will take like 5 minutes.

Dan as in Dan Abramov? Sounds awesome - if React wants to do half the work, why reinvent the wheel. Maybe that will help with code splitting of components? If you can render a component tree in the background (enough so that RUC downloads all the chunks) then maybe its possible to trigger that during prefetching but only actually commit the change when/if the non-preload route action is dispatched?

If I understand right, suspense is coming in React 17? It would be a pity to make adjusting for all these [breaking changes](https://github.com/facebook/react/milestone/40) a blocker for using rudy-respond.

File: src/pure-utils/actionToPath.js [hunk 1]
```diff
@@ -32,22 +29,29 @@ export default (
...
+    sluggifedParams[key] = transformSegment(segment, route, key)
     return sluggifedParams
   }, {})
+
+const transformSegment = (segment: string, route: Route, key: string) => {
+  if (typeof route.toPath === 'function') {
+    return route.toPath(segment, key)
+  }
+  else if (typeof segment === 'string') {
+    if (segment.indexOf('/') > -1) {
```
Comment (leggomuhgreggo) [inline src/pure-utils/actionToPath.js]: Shout out to the better ergonomics/readability of `.includes`

File: src/pure-utils/actionToPath.js [hunk 2]
```diff
@@ -32,22 +29,29 @@ export default (
...
-      }
-      else if (typeof route.toPath === 'function') {
-        sluggifedParams[key] = route.toPath(params[key], key)
-      }
-      else if (typeof params[key] === 'string') {
-        sluggifedParams[key] = params[key]
-      }
-    }
-
+    const segment = params[key]
```
Comment (leggomuhgreggo) [inline src/pure-utils/actionToPath.js]: might consider `Object.entries` if both the key and value are important -- avoids reference to object

Comment (leggomuhgreggo) [reply src/pure-utils/actionToPath.js]: Also Im not sure "segment" is the clearest name, but perhaps Im just not familiar with it's use in reference to param values.

**Comment by ScriptedAlchemy** [comment]
@leggomuhgreggo thanks for the review here!

**Comment by hedgepigdaniel** [comment]
https://github.com/faceyspacey/redux-first-router/pull/290 pulls in the latest documentation fixes from master

**Comment by hedgepigdaniel** [comment]
https://github.com/faceyspacey/redux-first-router/pull/291/files should fix #285

**Comment by hedgepigdaniel** [comment]
Might be worth looking at the latest comments here https://github.com/faceyspacey/redux-first-router/issues/41#issuecomment-424299971 by @dawnmist - looks like the integer coersion issue is not fully fixed.

**Comment by ScriptedAlchemy** [comment]
Thanks man , will do

**Comment by ScriptedAlchemy** [comment]
Preparing to release this, working on failing tests

File: src/pure-utils/actionToPath.js [hunk 3]
```diff
@@ -32,22 +29,29 @@ export default (
...
     return sluggifedParams
   }, {})
+
+const transformSegment = (segment: string, route: Route, key: string) => {
+  if (typeof route.toPath === 'function') {
+    return route.toPath(segment, key)
+  }
+  else if (typeof segment === 'string') {
+    if (segment.indexOf('/') > -1) {
+      return segment.split('/')
```
[the remark this answers is not in the document]
Comment (ScriptedAlchemy) [reply src/pure-utils/actionToPath.js]: not sure, theres no test on this, but ill ask @faceyspacey

**Pull request merged**